pydocstringformatter/run.py

```diff
@@ -123,11 +113,117 @@ def get_enabled_formatters(self) -> dict[str, _formatting.Formatter]:
 
         return enabled
 
-    def format_files(self, filepaths: list[Path]) -> bool:
-        """Format a list of files."""
+    def format_file_tokens(
+        self, tokens: list[tokenize.TokenInfo], filename: Path
+    ) -> tuple[list[tokenize.TokenInfo], bool]:
+        """Format a list of tokens.
+
+        tokens: List of tokens to format.
+        filename: Name of the file the tokens are from.
+
+        Returns:
+            A tuple containing [1] the formatted tokens in a list
+            and a boolean indicating if the tokens were changed.
+
+        Raises:
+            RuntimeError:
```

Comment:
I'd prefer to make a `UnstableResultError` which inherits from `PydocstringFormatterError`.
See:

Reply:
Done